chainer/training/trainer.py

```diff
@@ -14,6 +14,16 @@
 from chainer.training import trigger as trigger_module
 
 
+# Select the best-resolution timer function
+try:
+    _get_time = time.perf_counter
```

Comment:
Is there any reason to prefer `getattr`?
IMO, `try..catch` way is

* Clean and straight-forward
* EAFP is more pythonic than LBYL
* Faster (speed is not critical in this case, though)